### DISCUSSION TITLE
GPU usage of drop edge

### DISCUSSION THREAD

**Discussion by songyesog2000** [pyg-team/pytorch_geometric#4054] [Q&A] [answered]
Sorry for asking the dumb question here since i am a newbee for programming. I met a slow running on GPU using drop edge in each layer of my nn model.  [Drop edge documentation here](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/dropout.html#dropout_adj). I wonder is this problem caused by the cuda compatibility of the function or the generation of the random bool value.

Thank you in advance for checking this for me and maintaining such an amazing package!
Sam

**Comment by rusty1s** [comment, marked as answer]
Thanks for reporting. There were indeed some issues that slowed down GNN execution (in particular when `force_undirected=True` set). I fixed this in https://github.com/pyg-team/pytorch_geometric/pull/4059.

**Reply by songyesog2000** [reply]
Thank you, really helpful!
